### ISSUE TITLE
Implement a Notification.GetMessage(ReportingDescriptor) extension method

### ISSUE THREAD

**Issue by michaelcfanning** [microsoft/sarif-sdk#1733]
.. analogous to Result.GetMessage(Run)

**Comment by eddynaka** [comment]
@michaelcfanning is this issue still valid?

**Comment by michaelcfanning** [comment]
Yes, sir. This topic is a little complicated. The first thing to note is that you may require new helpers to properly resolve the Notification.Descriptor (which is actually a reference to the notification reporting descriptor). See Core\Result.cs|GetRule() helper which provides something similar. The spec has all the logic, as usual for resolving everything. @lgolding is extremely helpful if you want assistance here.

With some additional helpers, the next complexity would be to consider how much of Result.GetMessage() logic is appropriate to share with a notification helper.

Thanks for all the contributions! They are much appreciated.
